functorch/_src/aot_autograd.py

```diff
@@ -167,86 +181,336 @@ def posthook(grad_input, grad_output):
+        #     return x_updated, x_updated.view(-1)
+        #
+        # However, this is still wrong: we can't return x_updated.view(-1) to the user. We are on the hook to return:
+        # def traced_forward(x):
+        #     x_updated = x.mul(2)
+        #     return x_updated
+        #
+        # def compiled_fn(x)
+        #     x_updated = traced_forward(x)
+        #     x.copy_(x_updated)
+        #     return x.view(-1)
+        #
+        # Why can't we return x_updated.view(-1) to the user?
+        # It can have different metadata from x.view(-1)! Specifically, the input x could be a non-memory-dense tensor,
+        # But the intermediate created by our graph, x_updated, will always be memory-dense.
+        def filter_and_record_aliased_outs(outputs):
+            # NOTE: this dict will clobber keys if we have multiple inputs that alias.
+            # Let's say inpA and inpB alias, and the user generated an output using out = inpA.view(...)
+            # For now, since we're not handling the case with multiple _base's sharing a storage,
+            # it is actually fine to arbitrarily pick which input to regenerate the aliased output from.
+            # e.g. out_new = inpB.as_strided(out.size(), out.stride(), out.storage_offset())
+            #
+            # This will be more complicated when you have multiple _base tensors aliasing the same
+            # underlying storage, when we eventually handle that.
+            # We'll need to ensure that we generate the view off of the right base.
+            inp_storage_refs = {StorageWeakRef(inpt.storage()): idx for idx, inpt in enumerate(flat_f_args)}
+            inp_tensor_ids = {id(inpt) for inpt in flat_f_args if isinstance(inpt, torch.Tensor)}
+            inp_storage_refs_set = set(inp_storage_refs)
+            non_aliased_outs = []
+            for o in outputs:
+                # Note: When detecting input/output aliasing, we NEED to do it using the outer FunctionalTensorWrapper objects.
+                # In the case where we mutate an input *and* return a view of it, the outer wrappers will still alias,
+                # but the inner tensors no longer alias.
+                if isinstance(o, torch.Tensor):
+                    out_storage_ref = StorageWeakRef(o.storage())
+                    # Note: if the function returns an output that *is* an input, we still cannot
+                    # return it in the graph.
+                    # Specifically, let's say:
+                    # - the input was mutated and returned
+                    # - the input was a view (._is_view() is true)
+                    # Then returning the output in the graph won't preserve the ._is_view() property,
+                    # because we'll be returning the "updated input" (which was created from an out-of-place op)
+                    if out_storage_ref in inp_storage_refs:
+                        maybe_aliased_inp_idx = inp_storage_refs[out_storage_ref]
```

Comment:
what does 'maybe' refer to? is it uncertain that aliasing occured?  or just not certain whether one or more inputs were aliased (due to the comment about _base above)